src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -235,27 +226,105 @@ private void turnOffReadiness(Throwable exception){
             throw new IllegalStateException("Failed to create RabbitMQ connection using configuration", e);
         }
 
-        this.connection.addBlockedListener(new BlockedListener() {
-            @Override
-            public void handleBlocked(String reason) throws IOException {
-                LOGGER.warn("RabbitMQ blocked connection: {}", reason);
+        addBlockedListenersToConnection(this.connection);
+        addRecoveryListenerToConnection(this.connection);
+    }
+
+    private void addShutdownListenerToChannel(Channel channel, Boolean withRecovery) {
+        channel.addShutdownListener(cause -> {
```

Comment:
Please add try...catch block for this method

Reply:
thrown inside a callback

Reply:
Yes, please catch exception in the call back